core/templates/pages/collection-editor-page/navbar/collection-editor-navbar-breadcrumb.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit test for collection editor navbar breadcrumb component.
+ */
+
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { TestBed, waitForAsync, ComponentFixture, fakeAsync } from '@angular/core/testing';
+import { CollectionEditorPageConstants } from '../collection-editor-page.constants';
+import { CollectionEditorNavbarBreadcrumbComponent } from './collection-editor-navbar-breadcrumb.component';
+import { CollectionEditorStateService } from '../services/collection-editor-state.service';
+import { CollectionEditorRoutingService } from '../services/collection-editor-routing.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { Collection } from 'domain/collection/collection.model';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+describe('Collection editor navbar breadcrumb component', () => {
+  let collectionEditorStateService: CollectionEditorStateService;
+  let collectionEditorRoutingService: CollectionEditorRoutingService;
+  let focusManagerService: FocusManagerService;
+  let component: CollectionEditorNavbarBreadcrumbComponent;
+  let fixture: ComponentFixture<CollectionEditorNavbarBreadcrumbComponent>;
+  let mockCollection: Collection;
+
+  let activeTab = 'edit';
+  let collectionData = {
+    id: 'collection_id',
+    title: 'collection title',
+    objective: 'collection objective',
+    languageCode: 'en',
+    tags: ['mock tag'],
+    playthrough: null,
+    category: 'Collection Category',
+    version: 0,
+    schemaVersion: 1,
+    nodes: []
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        CollectionEditorNavbarBreadcrumbComponent
+      ],
+      providers: [
+        CollectionEditorStateService,
+        CollectionEditorRoutingService,
+        FocusManagerService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      CollectionEditorNavbarBreadcrumbComponent);
+    component = fixture.componentInstance;
+    collectionEditorStateService = TestBed.inject(CollectionEditorStateService);
+    collectionEditorStateService = (collectionEditorStateService as unknown) as
+      jasmine.SpyObj<CollectionEditorStateService>;
+    collectionEditorRoutingService = TestBed.inject(
+      CollectionEditorRoutingService);
+    collectionEditorRoutingService = (
+      (collectionEditorRoutingService as unknown) as
+        jasmine.SpyObj<CollectionEditorRoutingService>);
+    focusManagerService = TestBed.inject(FocusManagerService);
+    focusManagerService = (focusManagerService as unknown) as
+      jasmine.SpyObj<FocusManagerService>;
+  });
+
+  beforeEach(() => {
+    mockCollection = new Collection(
```

Comment:
Lets not call this a mock. It is just an example one you create right? Just call it expectedCollection or similar..

Reply:
done